### PR TITLE
Settings page: Update secret field types from text to password

### LINKED ISSUE #1393: Settings Page: Convert the API Secret to a password field

### PR DESCRIPTION
**Pull request by mehmoodak**
## Description
Currently we shows secrets in text input field so now we are updating the field type to password.

Closes: #1393 

## How has this been tested?
- Ran existing tests

## Screenshots
image

### CODE REVIEW

File: src/UI/class-settings-page.php [hunk 1]
```diff
@@ -345,9 +359,13 @@ private function initialize_basic_section(): void {
...
-			'label_for'  => $field_id,
+			'option_key'    => $field_id,
+			'help_text'     => __( 'Your API secret is your secret code to <a href="https://www.parse.ly/help/api/analytics/">access our API</a>. It can be found at <code>dash.parsely.com/<var>yoursitedomain</var>/settings/api</code> (replace <var>yoursitedomain</var> with your domain name, e.g. <samp>mydomain.com</samp>).<br />If you haven\'t purchased access to the API and would like to do so, email your account manager or <a href="mailto:support@parsely.com">support@parsely.com</a>.', 'wp-parsely' ),
+			'label_for'     => $field_id,
+			'optional_args' => array(
+				'type'                => 'password',
+				'is_obfuscated_value' => true,
```
Comment (mehmoodak) [inline src/UI/class-settings-page.php:244]: Just added these two args .... rest are spacing changes

File: src/UI/class-settings-page.php [hunk 2]
```diff
@@ -714,22 +736,24 @@ public function print_text_tag( $args ): void {
...
+		echo sprintf( "<input type='%s' name='%s' id='%s' value='%s'", esc_attr( $type ), esc_attr( $name ), esc_attr( $id ), esc_attr( $value ) );
 
 		if ( isset( $args['help_text'] ) ) {
 			echo ' aria-describedby="' . esc_attr( $id ) . '-description"';
 		}
 
 		foreach ( $optional_args as $key => $val ) {
 			if ( \in_array( $key, $accepted_args, true ) ) {
-				echo ' ' . esc_attr( $key ) . '="' . esc_attr( $val ) . '"';
+				echo ' ' . esc_attr( $key ) . '="' . esc_attr( (string) $val ) . '"';
```
Comment (mehmoodak) [inline src/UI/class-settings-page.php:686]: To remove PHPStan error.

File: src/UI/class-settings-page.php [hunk 3]
```diff
@@ -1058,8 +1083,6 @@ public function validate_options( $input ) {
 			$input['disable_amp'] = 'true' === $input['disable_amp'];
 		}
 
-		$input['api_secret'] = sanitize_text_field( $input['api_secret'] );
```
Comment (mehmoodak) [inline src/UI/class-settings-page.php]: Duplicated so removed.

**Review by mehmoodak — COMMENTED** [review]
Added comments for easier review.

**Comment by acicovic** [comment]
I [pushed a commit here](https://github.com/Parsely/wp-parsely/pull/1418/commits/00f50945a854a82da6331428bd38ffdc7f6d7606). Now the obfuscation respects the user’s input length (which wasn’t the case before and could be a bit confusing). I’ve manually tested this on my end multiple times, but I’d appreciate others checking it, or even testing it if possible.

**Comment by mehmoodak** [comment]
Change looks good 👍 (cannot approve my own PR so just leaving a comment)

**Review by acicovic — APPROVED** [review]
Great, this works for me!

We can ignore the SonarCloud warnings.

**Comment by acicovic** [comment]
@mehmoodak, due to the recent issues, some of your commits weren't signed, which leads to merging being blocked. But this is OK, let's merge this. I can override it if you can't for any reason.

**Comment by mehmoodak** [comment]
> some of your commits weren't signed, which leads to merging being blocked.

Resolved this issue and now going ahead with the merge.

**Pull request merged**